From b6f3b4addd83f4b28e7e6d98e0d6cfbeb351f370 Mon Sep 17 00:00:00 2001
From: Sridharan S N <quic_sridsn@quicinc.com>
Date: Wed, 6 Dec 2023 12:52:36 +0530
Subject: [PATCH 393/500] PCI: qcom: Set max payload size 256 bytes for
 IPQ9574/IPQ5332

This patch sets 256 bytes as payload size for IPQ9574/IPQ5332.
This allows PCIe RC to use the max payload size when a capable
link partner is connected.

Change-Id: I3c58c54635b58892d52e7963df4339aca049d46a
Signed-off-by: Sridharan S N <quic_sridsn@quicinc.com>
---
 drivers/pci/controller/dwc/pcie-qcom.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/pci/controller/dwc/pcie-qcom.c b/drivers/pci/controller/dwc/pcie-qcom.c
index 4fabcf707a18..1321efef7c26 100644
--- a/drivers/pci/controller/dwc/pcie-qcom.c
+++ b/drivers/pci/controller/dwc/pcie-qcom.c
@@ -1109,6 +1109,11 @@ static int qcom_pcie_post_init(struct qcom_pcie *pcie)
 	u32 val;
 	int i;
 
+	val = readl(pci->dbi_base + offset + PCI_EXP_DEVCTL);
+	val &= ~PCI_EXP_DEVCTL_PAYLOAD;
+	val |= PCI_EXP_DEVCTL_PAYLOAD_256B;
+	writel(val, pci->dbi_base + offset + PCI_EXP_DEVCTL);
+
 	val = readl(pcie->parf + PARF_PHY_CTRL);
 	val &= ~PHY_TEST_PWR_DOWN;
 	writel(val, pcie->parf + PARF_PHY_CTRL);
-- 
2.34.1

